tests/MongoDB.Driver.Core.Tests/Specifications/connection-string/tests/invalid-uris.yml

```diff
@@ -191,3 +191,12 @@ tests:
         hosts: ~
         auth: ~
         options: ~
+    -
+        description: "Username with password containing an unescaped percent sign"
+        uri: "mongodb://alice%foo:bar@127.0.0.1"
+        valid: false
+        warning: ~
+        hosts: ~
+        auth: ~
+        options: ~        
+    
```

Comment:
Possibly not in the scope of this ticket, but Visual Studio is not happy with the `connection-string\TestRunner` class. It won't let me select and debug individual tests unless I make the following changes to use the new `JsonDrivenTestCase` class which is Visual Studio/Xunit friendly:

Replace:

    public void RunTestDefinition(BsonDocument definition)

With:

    public void RunTestDefinition(JsonDrivenTestCase testCase)
    {
        var definition = testCase.Test;

And replace the entire `TestCaseFactory` class with:

    private class TestCaseFactory : JsonDrivenTestCaseFactory
    {
        private static readonly string[] __ignoredTestNames = new string[]
        {
            "invalid-uris.json:Missing delimiting slash between hosts and options"
        };

        protected override string PathPrefix => "MongoDB.Driver.Core.Tests.Specifications.connection_string.tests.";

        protected override IEnumerable<JsonDrivenTestCase> CreateTestCases(BsonDocument document)
        {
            return base.CreateTestCases(document)
                .Where(test => !__ignoredTestNames.Any(ignoredName => test.Name.EndsWith(ignoredName)));
        }
    }

Reply:
I've gone ahead and made this change, but we can revert it and leave it for another ticket if you think that's more appropriate.

Reply:
I'm OK with it here as part of this ticket as long as new one else objects.

Reply:
SGTM!

Reply:
We were also seeing issues on Rider